server/go/pkg/server/database.go

```diff
@@ -6,6 +6,27 @@ import (
 	"github.com/go-sql-driver/mysql"
 )
 
+var (
+	dbconnection *sql.DB
+)
+
+func getMysql(connStr string, pool bool) (*sql.DB, error) {
+	if !pool {
+		return newMysql(connStr)
+	}
+
+	if (*sql.DB)(nil) == dbconnection {
```

Comment:
a vanilla `nil` check will suffice as long as we never store `(*sql.DB)(nil)` in the interface var. Should also add a basic unit test to cover the new logic (time to create a database_test.go perhaps?).